APIServer.js

```diff
@@ -25,25 +46,38 @@ exports.checkUrl = function (req, res) {
 
     } else {
 
-        //Save the token to notify the WStore
-        if (apiKey) {
+        var path = url.parse(bodyUrl).pathname;
 
-            db.addToken(apiKey, function (err) {
-                if (err) {
-                    logger.error(err);
-                }
-            });
-        }
+        isAdmin(req.user.id, path, function (err, isAdmin) {
```

Comment:
The method isAdmin is correct, but you cannot execute it before checking the path. Note that in case you are using an absolute URL without query strings this works (pathname not include the query string part). However, if you are making the request to a service registered with a relative path getAdmins is not going to return anything. You have to first check the path and determine the effective external path, once you know that the service exists and what is it external path you can call the isAdmin method